--- avr_audio_bootloader/fsk/encoder.py	2026-08-22 06:08:30
+++ avr_audio_bootloader/fsk/encoder.py	2026-08-22 06:10:25
@@ -66,11 +66,11 @@
   def _code_packet(self, data):
     assert len(data) <= self._packet_size
     if len(data) != self._packet_size:
-      data = data + '\x00' * (self._packet_size - len(data))
+      data = data + b'\x00' * (self._packet_size - len(data))
 
     crc = zlib.crc32(data) & 0xffffffff
 
-    data = map(ord, data)
+    data = list(data)
     crc_bytes = [crc >> 24, (crc >> 16) & 0xff, (crc >> 8) & 0xff, crc & 0xff]
     bytes = [0x55] * 4 + data + crc_bytes
     
@@ -84,11 +84,11 @@
     return self._encode(symbol_stream)
 
   def code(self, data, page_size=1024, blank_duration=0.06):
-    yield numpy.zeros((1.0 * self._sr, 1)).ravel()
+    yield numpy.zeros((int(self._sr), 1)).ravel()
     yield self._code_blank(1.0)
     if len(data) % page_size != 0:
       tail = page_size - (len(data) % page_size)
-      data += '\xff' * tail
+      data += b'\xff' * tail
     
     offset = 0
     remaining_bytes = len(data)
@@ -97,7 +97,7 @@
       size = min(remaining_bytes, self._packet_size)
       yield self._code_packet(data[offset:offset+size])
       num_packets_written += 1
-      if num_packets_written == page_size / self._packet_size:
+      if num_packets_written == page_size // self._packet_size:
         yield self._code_blank(blank_duration)
         num_packets_written = 0
       remaining_bytes -= size
@@ -165,7 +165,7 @@
       metavar='FILE')
   
   options, args = parser.parse_args()
-  data = file(args[0], 'rb').read()
+  data = open(args[0], 'rb').read()
   if len(args) != 1:
     logging.fatal('Specify one, and only one firmware .bin file!')
     sys.exit(1)
--- avr_audio_bootloader/audio_stream_writer.py	2026-08-22 06:08:30
+++ avr_audio_bootloader/audio_stream_writer.py	2026-08-22 06:09:02
@@ -40,7 +40,7 @@
     self._sample_rate = sample_rate
     self._bitdepth = bitdepth
     self._num_channels = num_channels
-    self._f = file(file_name, 'wb')
+    self._f = open(file_name, 'wb')
     self._num_bytes = 0
     self._write_header(False)
   
@@ -72,7 +72,7 @@
       scale = (1 << (bitdepth - 1)) - 1
       # pylint: disable-msg=C6407
       scaled_signal = scaled_signal * scale
-      scaled_signal = numpy.array(scaled_signal, dtype='i%d' % (bitdepth / 8))
+      scaled_signal = numpy.array(scaled_signal, dtype='i%d' % (bitdepth // 8))
 
     return scaled_signal
   
@@ -85,11 +85,11 @@
     
     current_position = f.tell()
     f.seek(0)
-    f.write('RIFF')
+    f.write(b'RIFF')
     f.write(struct.pack('<L', total_size))
-    f.write('WAVEfmt ')
-    bitrate = self._sample_rate * self._num_channels * (self._bitdepth / 8)
-    bits_per_sample = self._num_channels * (self._bitdepth / 8)
+    f.write(b'WAVEfmt ')
+    bitrate = self._sample_rate * self._num_channels * (self._bitdepth // 8)
+    bits_per_sample = self._num_channels * (self._bitdepth // 8)
     f.write(struct.pack(
         '<LHHLLHH',
         16,
@@ -99,7 +99,7 @@
         bitrate,
         bits_per_sample,
         self._bitdepth))
-    f.write('data')
+    f.write(b'data')
     f.write(struct.pack('<L', self._num_bytes))
     if restore_position:
       f.seek(current_position)
